packages/jellyfish-api-core/__tests__/category/loan/takeLoan.test.ts

```diff
@@ -3,279 +3,803 @@ import { GenesisKeys } from '@defichain/testcontainers'
+    timestamp = Math.floor(new Date().getTime() / 1000)
+    await alice.rpc.oracle.setOracleData(
+      oracleId,
+      timestamp,
+      {
+        prices: [
+          { tokenAmount: '1@DFI', currency: 'USD' },
+          { tokenAmount: '10000@BTC', currency: 'USD' },
+          { tokenAmount: '2@TSLA', currency: 'USD' },
+          { tokenAmount: '4@GOOGL', currency: 'USD' }
+        ]
+      })
+    await alice.generate(1)
+
+    // collateral token
+    await alice.rpc.loan.setCollateralToken({
+      token: 'DFI',
+      factor: new BigNumber(1),
+      fixedIntervalPriceId: 'DFI/USD'
+    })
+    await alice.generate(1)
+
+    await alice.rpc.loan.setCollateralToken({
+      token: 'BTC',
+      factor: new BigNumber(0.5),
+      fixedIntervalPriceId: 'BTC/USD'
+    })
+    await alice.generate(1)
+
+    // loan token
+    await alice.rpc.loan.setLoanToken({
+      symbol: 'TSLA',
+      fixedIntervalPriceId: 'TSLA/USD'
+    })
+    await alice.generate(1)
+
+    await alice.rpc.loan.setLoanToken({
+      symbol: 'GOOGL',
+      fixedIntervalPriceId: 'GOOGL/USD'
+    })
+    await alice.generate(1)
+
+    // loan scheme set up
+    await alice.rpc.loan.createLoanScheme({
+      minColRatio: 200,
+      interestRate: new BigNumber(3),
+      id: 'scheme'
+    })
+    await alice.generate(1)
     await tGroup.waitForSync()
-    const timestamp = Math.floor(new Date().getTime() / 1000)
-    await tGroup.get(0).rpc.oracle.setOracleData(oracleId, timestamp, { prices: [{ tokenAmount: '1@DFI', currency: 'USD' }, { tokenAmount: '10000@BTC', currency: 'USD' }, { tokenAmount: '2@TSLA', currency: 'USD' }] })
-    await tGroup.get(0).generate(1)
+
+    bobVaultAddr = await bob.generateAddress()
+    bobVaultId = await bob.rpc.loan.createVault({
+      ownerAddress: bobVaultAddr,
+      loanSchemeId: 'scheme'
+    })
+    await bob.generate(1)
     await tGroup.waitForSync()
 
+    // deposit on active vault
+    await alice.rpc.loan.depositToVault({
+      vaultId: bobVaultId, from: aliceAddr, amount: '10000@DFI'
+    })
+    await alice.generate(1)
+
+    await alice.rpc.loan.depositToVault({
+      vaultId: bobVaultId, from: aliceAddr, amount: '1@BTC'
+    })
+    await alice.generate(1)
+    await tGroup.waitForSync()
+
+    bobVault = await bob.rpc.loan.getVault(bobVaultId) as VaultActive
+    expect(bobVault.loanSchemeId).toStrictEqual('scheme')
+    expect(bobVault.ownerAddress).toStrictEqual(bobVaultAddr)
+    expect(bobVault.state).toStrictEqual('active')
+    expect(bobVault.collateralAmounts).toStrictEqual(['10000.00000000@DFI', '1.00000000@BTC'])
+    expect(bobVault.collateralValue).toStrictEqual(new BigNumber(15000))
+    expect(bobVault.loanAmounts).toStrictEqual([])
+    expect(bobVault.loanValue).toStrictEqual(new BigNumber(0))
+    expect(bobVault.interestAmounts).toStrictEqual([])
+    expect(bobVault.interestValue).toStrictEqual(new BigNumber(0))
+    expect(bobVault.collateralRatio).toStrictEqual(-1) // empty loan
```

Comment:
This should be `new BigNumber(-1)`. You may need to change the return type for `collateralRatio` in `getVault` RPC

Reply:
hmm.. have `collateralRatio` got decimal??

Reply:
LOL. Okay, then make sense to keep as number.